Improve search entry sizing

germanfr opened the pull request:
Let themes define horizontal paddings. Improve width calculations instead of multiplying it by 0.95.

An example of improvement (left before, right after):
![screenshot from 2017-03-06 00-24-13](https://cloud.githubusercontent.com/assets/10391266/23592615/0570954c-0204-11e7-9860-dbebb1b4ea0d.png)
 
In general, themes are not affected by this change or they have improved padding glitches caused by the applet itself.

There must be a better way of setting widths than with CSS styles, I guess.

ghost commented:
@germanfr Seems like a good idea 👍

germanfr commented:
I found a cleaner an easier solution without CSS styles that uses the Clutter method to set width instead. I created another commit, in case you wanted to see the old one (squash them or ask me to rebase if you don't want two commits).

NikoKrause commented:
I never saw something like that in your screenshot.
How can I reproduce that? A specific theme?

germanfr commented:
Try New-Minty or Railway (both in the spices).

I'm the developer of Railway and no matter what I do it looks like in the first screenshot. It only has width, padding and border-image, nothing more. I don't know the difference between the ones who don't have a problem and those who do, but this seems to be what is causing the issue. I think hardcoding things can't be any good.

NikoKrause commented on CinnVIIStarkMenu@NikoKrause/3.2/applet.js:
```diff
@@ -3609,9 +3616,7 @@ MyApplet.prototype = {
-        this.searchBox = new St.BoxLayout({ style_class: 'menu-search-box' });
-        this.searchBox.add_style_class_name("starkmenu-search-box");
-        this.searchBox.set_style("padding-right: 0px;padding-left: 0px;height:26px;");
+        this.searchBox = new St.BoxLayout({ style_class: 'starkmenu-search-box' });
```
This would be better:
```js
this.searchBox = new St.BoxLayout({ style_class: 'menu-search-box' });
this.searchBox.add_style_class_name("starkmenu-search-box");
```

germanfr replied:
Yes, the problem with that is that most themes add a big left padding on there and it looks awkward in the stark menu.
I removed that class because the search bar doesn't need any shared styles from the menu, a priori.

Like they are the same object (already styled) but at different locations so they need different paddings and such. The search bar itself still contains their classes, that's for the box containing it which is different.

NikoKrause commented:
Sorry. Yes. I saw this after I wrote the comment.
Ignore it.

Can you solve the conflict. Then I'll merge the PR.
Thanks.

germanfr commented:
Do you want me to squash both commits into one too?

NikoKrause commented:
Yeah, if you have done that before.
If not, I can squash it while merging.

Forgotten to mention: Can you change this in the `2.6/applet.js` file as well.
Those are changes, which won't break the menu in previous versions.

germanfr commented:
OK, then let me create a one commit with the changes in both and without conflicts.

germanfr committed:
Improve search entry sizing

Let themes define horizontal paddings. Improve width calculations
instead of multiplying parent width by 0.95.

germanfr commented:
Done. I manually replicated the changes into 2.6 (I think I didn't miss anything), but I haven't tested if it works because I don't have a 2.6 cinnamon.

NikoKrause commented:
Thanks
